src/aes.rs

```diff
@@ -33,17 +33,17 @@ const SBOX: SBox = SBox([
     0xBA, 0x78, 0x25, 0x2E, 0x1C, 0xA6, 0xB4, 0xC6, 0xE8, 0xDD, 0x74, 0x1F, 0x4B, 0xBD, 0x8B, 0x8A,
     0x70, 0x3E, 0xB5, 0x66, 0x48, 0x03, 0xF6, 0x0E, 0x61, 0x35, 0x57, 0xB9, 0x86, 0xC1, 0x1D, 0x9E,
     0xE1, 0xF8, 0x98, 0x11, 0x69, 0xD9, 0x8E, 0x94, 0x9B, 0x1E, 0x87, 0xE9, 0xCE, 0x55, 0x28, 0xDF,
-    0x8C, 0xA1, 0x89, 0x0D, 0xBF, 0xE6, 0x42, 0x68, 0x41, 0x99, 0x2D, 0x0F, 0xB0, 0x54, 0xBB, 0x16,
-]);
+    0x8C, 0xA1, 0x89, 0x0D, 0xBF, 0xE6, 0x42, 0x68, 0x41, 0x99, 0x2D, 0x0F, 0xB0, 0x54, 0xBB, 0x16
+]));
 
-const RCON: RCon = RCon([
-    0x8d, 0x01, 0x02, 0x04, 0x08, 0x10, 0x20, 0x40, 0x80, 0x1b, 0x36,
-]);
+const RCON: RCon = RCon(secret_bytes!([
+    0x8d, 0x01, 0x02, 0x04, 0x08, 0x10, 0x20, 0x40, 0x80, 0x1b, 0x36
+]));
 
 fn sub_bytes(state: Block) -> Block {
     let mut st = state;
     for i in 0..16 {
-        st[i] = SBOX[state[i]];
+        st[i] = SBOX[U8::declassify(state[i]) as usize];
```

Comment:
The `as usize` really shouldn't be needed. That's too Rusty...
I'm also not super fond of the `U8::declassify` but I think its ok for now. We might have to think about it though.

Reply:
To get rid of the `as usize`, you have to allow indexing of the arrays by `u8` and `u32`. I did it but then it breaks type inference in other places because before arrays could only be indexed by one type of integer. Specifficaly, look at blake2b.rs:94 that I had to add because of this change.

Concerning the `declassify`, I strongly support that we maintain it: it is here because the S-box-based implementation of AES is not constant-time. We remind the user of the non-constant-timedness of her implementation by requiring to explicitely declassify if needed.

Reply:
Ok, then please add a TODO for the `as usize` so we can fix that.
I'm fine with keeping the `declassify` I was just pondering the question how we could make this nicer... Maybe add an issue or todo to figure out how to make the declassification nicer :slightly_smiling_face: